Mismatch in numerical precision on the latitudes of the CESM ensemble members (34, 35 and 101-105) for atmospheric variables

It appears that the numerical precision on the latitudes of the last 7 ensemble members (numbers 34, 35 and 101-105) for atmospheric variables is actually different than on the first 33 members.  

This causes problems when trying to use xarray to concatenate the members together into a single dataset, as xarray would try to merge both of the latitude arrays and effectively double the size of the latitude dimension. 

```python
In [12]: z = xr.open_dataset("/glade/collections/cdg/data/cesmLE/CESM-CAM5-BGC-LE/atm/proc/tseries/monthly/TS/b.e11.B20TRC5CNBDRD.f09_g16.003.cam.h
    ...: 0.TS.192001-200512.nc").lat
    ...:

In [13]: x = xr.open_dataset("/glade/collections/cdg/data/cesmLE/CESM-CAM5-BGC-LE/atm/proc/tseries/monthly/TS/b.e11.B20TRC5CNBDRD.f09_g16.105.cam.h
    ...: 0.TS.192001-200512.nc").lat

In [14]: y = xr.open_dataset("/glade/collections/cdg/data/cesmLE/CESM-CAM5-BGC-LE/atm/proc/tseries/monthly/TS/b.e11.B20TRC5CNBDRD.f09_g16.104.cam.h
    ...: 0.TS.192001-200512.nc").lat

In [15]: xr.testing.assert_identical(x, y)

In [16]: xr.testing.assert_identical(z, y)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-16-a0aa58e132f5> in <module>
----> 1 xr.testing.assert_identical(z, y)

/glade/work/abanihi/softwares/miniconda3/envs/analysis/lib/python3.7/site-packages/xarray/testing.py in assert_identical(a, b)
     88     elif isinstance(a, DataArray):
     89         assert a.name == b.name
---> 90         assert a.identical(b), formatting.diff_array_repr(a, b, 'identical')
     91     elif isinstance(a, (Dataset, Variable)):
     92         assert a.identical(b), formatting.diff_dataset_repr(a, b, 'identical')

AssertionError: Left and right DataArray objects are not identical

Differing values:
L
    array([-90.      , -89.057592, -88.115183, -87.172775, -86.230366, -85.287958,
           -84.34555 , -83.403141, -82.460733, -81.518325, -80.575916, -79.633508,
           -78.691099, -77.748691, -76.806283, -75.863874, -74.921466, -73.979058,
           -73.036649, -72.094241, -71.151832, -70.209424, -69.267016, -68.324607,
           -67.382199, -66.439791, -65.497382, -64.554974, -63.612565, -62.670157,
           -61.727749, -60.78534 , -59.842932, -58.900524, -57.958115, -57.015707,
           -56.073298, -55.13089 , -54.188482, -53.246073, -52.303665, -51.361257,
           -50.418848, -49.47644 , -48.534031, -47.591623, -46.649215, -45.706806,
           -44.764398, -43.82199 , -42.879581, -41.937173, -40.994764, -40.052356,
           -39.109948, -38.167539, -37.225131, -36.282723, -35.340314, -34.397906,
           -33.455497, -32.513089, -31.570681, -30.628272, -29.685864, -28.743455,
           -27.801047, -26.858639, -25.91623 , -24.973822, -24.031414, -23.089005,
           -22.146597, -21.204188, -20.26178 , -19.319372, -18.376963, -17.434555,
           -16.492147, -15.549738, -14.60733 , -13.664921, -12.722513, -11.780105,
           -10.837696,  -9.895288,  -8.95288 ,  -8.010471,  -7.068063,  -6.125654,
            -5.183246,  -4.240838,  -3.298429,  -2.356021,  -1.413613,  -0.471204,
             0.471204,   1.413613,   2.356021,   3.298429,   4.240838,   5.183246,
             6.125654,   7.068063,   8.010471,   8.95288 ,   9.895288,  10.837696,
            11.780105,  12.722513,  13.664921,  14.60733 ,  15.549738,  16.492147,
            17.434555,  18.376963,  19.319372,  20.26178 ,  21.204188,  22.146597,
            23.089005,  24.031414,  24.973822,  25.91623 ,  26.858639,  27.801047,
            28.743455,  29.685864,  30.628272,  31.570681,  32.513089,  33.455497,
            34.397906,  35.340314,  36.282723,  37.225131,  38.167539,  39.109948,
            40.052356,  40.994764,  41.937173,  42.879581,  43.82199 ,  44.764398,
            45.706806,  46.649215,  47.591623,  48.534031,  49.47644 ,  50.418848,
            51.361257,  52.303665,  53.246073,  54.188482,  55.13089 ,  56.073298,
            57.015707,  57.958115,  58.900524,  59.842932,  60.78534 ,  61.727749,
            62.670157,  63.612565,  64.554974,  65.497382,  66.439791,  67.382199,
            68.324607,  69.267016,  70.209424,  71.151832,  72.094241,  73.036649,
            73.979058,  74.921466,  75.863874,  76.806283,  77.748691,  78.691099,
            79.633508,  80.575916,  81.518325,  82.460733,  83.403141,  84.34555 ,
            85.287958,  86.230366,  87.172775,  88.115183,  89.057592,  90.      ])
R
    array([-90.      , -89.057592, -88.115183, -87.172775, -86.230366, -85.287958,
           -84.34555 , -83.403141, -82.460733, -81.518325, -80.575916, -79.633508,
           -78.691099, -77.748691, -76.806283, -75.863874, -74.921466, -73.979058,
           -73.036649, -72.094241, -71.151832, -70.209424, -69.267016, -68.324607,
           -67.382199, -66.439791, -65.497382, -64.554974, -63.612565, -62.670157,
           -61.727749, -60.78534 , -59.842932, -58.900524, -57.958115, -57.015707,
           -56.073298, -55.13089 , -54.188482, -53.246073, -52.303665, -51.361257,
           -50.418848, -49.47644 , -48.534031, -47.591623, -46.649215, -45.706806,
           -44.764398, -43.82199 , -42.879581, -41.937173, -40.994764, -40.052356,
           -39.109948, -38.167539, -37.225131, -36.282723, -35.340314, -34.397906,
           -33.455497, -32.513089, -31.570681, -30.628272, -29.685864, -28.743455,
           -27.801047, -26.858639, -25.91623 , -24.973822, -24.031414, -23.089005,
           -22.146597, -21.204188, -20.26178 , -19.319372, -18.376963, -17.434555,
           -16.492147, -15.549738, -14.60733 , -13.664921, -12.722513, -11.780105,
           -10.837696,  -9.895288,  -8.95288 ,  -8.010471,  -7.068063,  -6.125654,
            -5.183246,  -4.240838,  -3.298429,  -2.356021,  -1.413613,  -0.471204,
             0.471204,   1.413613,   2.356021,   3.298429,   4.240838,   5.183246,
             6.125654,   7.068063,   8.010471,   8.95288 ,   9.895288,  10.837696,
            11.780105,  12.722513,  13.664921,  14.60733 ,  15.549738,  16.492147,
            17.434555,  18.376963,  19.319372,  20.26178 ,  21.204188,  22.146597,
            23.089005,  24.031414,  24.973822,  25.91623 ,  26.858639,  27.801047,
            28.743455,  29.685864,  30.628272,  31.570681,  32.513089,  33.455497,
            34.397906,  35.340314,  36.282723,  37.225131,  38.167539,  39.109948,
            40.052356,  40.994764,  41.937173,  42.879581,  43.82199 ,  44.764398,
            45.706806,  46.649215,  47.591623,  48.534031,  49.47644 ,  50.418848,
            51.361257,  52.303665,  53.246073,  54.188482,  55.13089 ,  56.073298,
            57.015707,  57.958115,  58.900524,  59.842932,  60.78534 ,  61.727749,
            62.670157,  63.612565,  64.554974,  65.497382,  66.439791,  67.382199,
            68.324607,  69.267016,  70.209424,  71.151832,  72.094241,  73.036649,
            73.979058,  74.921466,  75.863874,  76.806283,  77.748691,  78.691099,
            79.633508,  80.575916,  81.518325,  82.460733,  83.403141,  84.34555 ,
            85.287958,  86.230366,  87.172775,  88.115183,  89.057592,  90.      ])
Differing coordinates:
L * lat      (lat) float64 -90.0 -89.06 -88.12 -87.17 ... 87.17 88.12 89.06 90.0
    long_name: latitude
    units: degrees_north
R * lat      (lat) float64 -90.0 -89.06 -88.12 -87.17 ... 87.17 88.12 89.06 90.0
    long_name: latitude
    units: degrees_north

```

![Screen Shot 2019-07-17 at 1 32 44 PM](https://user-images.githubusercontent.com/13301940/61405585-6801fe80-a897-11e9-825f-2bcb3cdde8b2.png)
![Screen Shot 2019-07-17 at 1 32 33 PM](https://user-images.githubusercontent.com/13301940/61405590-6b958580-a897-11e9-9688-99c18914a24d.png)


One possible solution is to  overwrite the latitude coordinate array on all files to be the same thing (drawn from the first member) before doing the xarray concatenation at runtime.

Cc @matt-long, @kmpaul, @jukent

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -36,7 +36,8 @@ Internal Changes
 
 - Refactor existing functionality to make intake-esm robust and extensible. (:pr:`77`) `Anderson Banihirwe`_
 
-
+- Add ``aggregate._override_coords`` function to override dim coordinates except time
+  in case there's floating point precision difference. (:pr:`108`) `Anderson Banihirwe`_
 
 Intake-esm v2019.5.11 (2019-05-11)
 ===================================

--- a/intake_esm/aggregate.py
+++ b/intake_esm/aggregate.py
@@ -26,6 +26,10 @@ def ensure_time_coord_name(ds, time_coord_name_default):
 
 
 def _override_coords(dsets, time_coord_name):
+    """
+    Override coordinates except time by using coordinates from the first
+    dataset in the list.
+    """
     dim_coords_except_time = set(dsets[0].coords).intersection(set(dsets[0].dims)) - set(
         [time_coord_name]
     )
@@ -33,7 +37,7 @@ def _override_coords(dsets, time_coord_name):
     datasets = [dsets[0]]
     for ds in dsets[1:]:
         if not all(ds[name].shape == dim_sizes_first[name] for name in dim_coords_except_time):
-            raise RuntimeError('dataset coord mismatch')
+            raise AssertionError(f'Dataset coordinates mismatch: {dsets[0].dims} != {ds.dims}')
         datasets.append(ds.drop(dim_coords_except_time))
 
     return datasets

--- a/intake_esm/aggregate.py
+++ b/intake_esm/aggregate.py
@@ -25,6 +25,36 @@ def ensure_time_coord_name(ds, time_coord_name_default):
     )
 
 
+def _override_coords(dsets, time_coord_name):
+    dim_coords_except_time = set(dsets[0].coords).intersection(set(dsets[0].dims)) - set(
+        [time_coord_name]
+    )
+    dim_sizes_first = {name: da.shape for name, da in dsets[0].coords.items()}
+    datasets = [dsets[0]]
+    for ds in dsets[1:]:
+        if not all(ds[name].shape == dim_sizes_first[name] for name in dim_coords_except_time):
+            raise RuntimeError('dataset coord mismatch')
+        datasets.append(ds.drop(dim_coords_except_time))
+
+    return datasets
+
+
+def _drop_additional_dims(dsets):
+    all_dims = [set(dset.dims) for dset in dsets]
+    common_dims = all_dims[0].intersection(*all_dims[1:])
+    ds = []
+    for dset in dsets:
+        dset_dims = set(dset.dims)
+        keep_dims = dset_dims.intersection(common_dims)
+        drop_dims = list(dset_dims - keep_dims)
+        if drop_dims:
+            ds.append(dset.drop_dims(drop_dims))
+        else:
+            ds.append(dset)
+
+    return ds
+
+
 def dict_union(*dicts, merge_keys=['history', 'tracking_id'], drop_keys=[]):
     """Return the union of two or more dictionaries."""
     if len(dicts) > 2:
@@ -123,10 +153,12 @@ def concat_time_levels(dsets, time_coord_name_default, restore_non_dim_coords=Fa
             return dsets[0]
 
     attrs = dict_union(*[ds.attrs for ds in dsets])
+    time_coord_name = ensure_time_coord_name(dsets[0], time_coord_name_default)
 
+    # Equivalent to xr.align(*dsets, tolerance=xtol, exclude='time')
+    dsets = _override_coords(dsets, time_coord_name)
     # get static vars from first dataset
     first = dsets[0]
-    time_coord_name = ensure_time_coord_name(first, time_coord_name_default)
 
     def drop_unnecessary_coords(ds):
         """Drop coordinates that do not correspond with dimensions."""
@@ -170,12 +202,19 @@ def concat_ensembles(
     if member_ids is None:
         member_ids = np.arange(0, len(dsets))
 
+    time_coord_name = ensure_time_coord_name(dsets[0], time_coord_name_default)
+    dsets = _drop_additional_dims(dsets)
     attrs = dict_union(*[ds.attrs for ds in dsets])
 
     # align first to deal with the fact that some ensemble members have different lengths
     # inner join keeps only overlapping segments of each ensemble
     # outer join gives us the longest possible record
-    dsets_aligned = xr.align(*dsets, join=join)
+    dim_coords_except_time = set(dsets[0].coords).intersection(set(dsets[0].dims)) - set(
+        [time_coord_name]
+    )
+    dsets_aligned = xr.align(*dsets, join=join, exclude=dim_coords_except_time)
+    # Equivalent to xr.align(*dsets_aligned, tolerance=xtol, exclude='time')
+    dsets_aligned = _override_coords(dsets_aligned, time_coord_name)
 
     # use coords and static_vars from first dataset
     first = dsets_aligned[0]
@@ -201,6 +240,8 @@ def concat_ensembles(
 
 def set_coords(ds, varname):
     """Set all variables except varname to be coords."""
+    if isinstance(varname, str):
+        varname = [varname]
     coord_vars = set(ds.data_vars) - set(varname)
     return ds.set_coords(coord_vars)
 

--- a/intake_esm/aggregate.py
+++ b/intake_esm/aggregate.py
@@ -128,7 +128,9 @@ def _restore_non_dim_coords(ds):
     return ds
 
 
-def concat_time_levels(dsets, time_coord_name_default, restore_non_dim_coords=False):
+def concat_time_levels(
+    dsets, time_coord_name_default, restore_non_dim_coords=False, override_coords=False
+):
     """
     Concatenate datasets across "time" levels, taking time invariant variables
     from the first dataset.
@@ -141,6 +143,9 @@ def concat_time_levels(dsets, time_coord_name_default, restore_non_dim_coords=Fa
         Default name of the time coordinate
     restore_non_coord_dim : bool, default (False)
         Whether or not to restore non coord dims
+    override_coords: bool, default (False)
+        Whether or not to drop all coordinates associated with dimensions
+        (except time) from all but the first entry in dsets.
 
     Returns
     -------
@@ -160,7 +165,8 @@ def concat_time_levels(dsets, time_coord_name_default, restore_non_dim_coords=Fa
     time_coord_name = ensure_time_coord_name(dsets[0], time_coord_name_default)
 
     # https://github.com/NCAR/intake-esm/issues/104#issuecomment-513404844
-    dsets = _override_coords(dsets, time_coord_name)
+    if override_coords:
+        dsets = _override_coords(dsets, time_coord_name)
     # get static vars from first dataset
     first = dsets[0]
 
@@ -196,9 +202,37 @@ def concat_ensembles(
     join='inner',
     ensemble_dim_name='member_id',
     time_coord_name_default='time',
+    override_coords=False,
 ):
     """Concatenate datasets across an ensemble dimension, taking coordinates and
     time-invariant variables from the first ensemble member.
+
+    Parameters
+    ----------
+    dsets : list
+        A list of datasets to concatenate.
+
+    member_ids : list, default (None)
+         A list of ids for the ensemble members
+
+    join : str
+        Accepted values: ``inner``, ``outer``
+
+    ensemble_dim_name : str, default(``member_id``)
+       Ensemble dimension name to use.
+
+    time_coord_name_default : str
+        Default name of the time coordinate
+
+    override_coords: bool, default (False)
+        Whether or not to drop all coordinates associated with dimensions
+        (except time) from all but the first entry in dsets.
+
+    Returns
+    -------
+    dset : xarray.Dataset,
+        The concatenated dataset.
+
     """
     if len(dsets) == 1:
         return _restore_non_dim_coords(dsets[0])
@@ -219,7 +253,8 @@ def concat_ensembles(
 
     # https://github.com/NCAR/intake-esm/issues/104#issuecomment-513404844
     dsets_aligned = xr.align(*dsets, join=join, exclude=dim_coords_except_time)
-    dsets_aligned = _override_coords(dsets_aligned, time_coord_name)
+    if override_coords:
+        dsets_aligned = _override_coords(dsets_aligned, time_coord_name)
 
     # use coords and static_vars from first dataset
     first = dsets_aligned[0]

--- a/intake_esm/aggregate.py
+++ b/intake_esm/aggregate.py
@@ -27,8 +27,8 @@ def ensure_time_coord_name(ds, time_coord_name_default):
 
 def _override_coords(dsets, time_coord_name):
     """
-    Override coordinates except time by using coordinates from the first
-    dataset in the list.
+    Return a list of datasets where all coordinates associated with dimensions
+    (except time) have been dropped from all but the first entry.
     """
     dim_coords_except_time = set(dsets[0].coords).intersection(set(dsets[0].dims)) - set(
         [time_coord_name]
@@ -159,7 +159,7 @@ def concat_time_levels(dsets, time_coord_name_default, restore_non_dim_coords=Fa
     attrs = dict_union(*[ds.attrs for ds in dsets])
     time_coord_name = ensure_time_coord_name(dsets[0], time_coord_name_default)
 
-    # Equivalent to xr.align(*dsets, tolerance=xtol, exclude='time')
+    # https://github.com/NCAR/intake-esm/issues/104#issuecomment-513404844
     dsets = _override_coords(dsets, time_coord_name)
     # get static vars from first dataset
     first = dsets[0]
@@ -216,8 +216,9 @@ def concat_ensembles(
     dim_coords_except_time = set(dsets[0].coords).intersection(set(dsets[0].dims)) - set(
         [time_coord_name]
     )
+
+    # https://github.com/NCAR/intake-esm/issues/104#issuecomment-513404844
     dsets_aligned = xr.align(*dsets, join=join, exclude=dim_coords_except_time)
-    # Equivalent to xr.align(*dsets_aligned, tolerance=xtol, exclude='time')
     dsets_aligned = _override_coords(dsets_aligned, time_coord_name)
 
     # use coords and static_vars from first dataset

--- a/intake_esm/source.py
+++ b/intake_esm/source.py
@@ -57,6 +57,8 @@ class BaseSource(intake_xarray.base.DataSourceMixin):
             _kwargs.update(ensemble_dim_name='member_id')
         if 'chunks' not in _kwargs:
             _kwargs.update(chunks={_kwargs['time_coord_name']: 'auto'})
+        if 'override_coords' not in _kwargs:
+            _kwargs.update(override_coords=False)
         if 'join' not in _kwargs:
             _kwargs.update(join='outer')
         if 'preprocess' not in _kwargs:
@@ -103,13 +105,20 @@ class BaseSource(intake_xarray.base.DataSourceMixin):
                         for url in urlpath_ei_vi
                     ]
 
-                    var_dset_i = aggregate.concat_time_levels(dsets, kwargs['time_coord_name'])
+                    var_dset_i = aggregate.concat_time_levels(
+                        dsets,
+                        time_coord_name_default=kwargs['time_coord_name'],
+                        override_coords=kwargs['override_coords'],
+                    )
                     var_dsets.append(var_dset_i)
                 member_ids.append(m_id)
                 member_dset_i = aggregate.merge(dsets=var_dsets)
                 member_dsets.append(member_dset_i)
             _ds = aggregate.concat_ensembles(
-                member_dsets, member_ids=member_ids, join=kwargs['join']
+                member_dsets,
+                member_ids=member_ids,
+                join=kwargs['join'],
+                override_coords=kwargs['override_coords'],
             )
             all_dsets[dset_id] = _ds
 

--- a/tests/test_aggregate.py
+++ b/tests/test_aggregate.py
@@ -38,7 +38,7 @@ def test_merge(dsets):
 
 
 def test_concat_time_levels(dsets):
-    dsets[0]['time'].data = (dsets[0].time + (2 * 2920.0)).data
+    dsets[0]['time'].data = dsets[0].time.data + (2 * 2920.0)
     ds = aggregate.concat_time_levels(dsets, 'time')
     assert ds.time.shape == (5840,)
 
@@ -50,11 +50,15 @@ def test_concat_ensembles(dsets):
 
 
 def test_concat_ensembles_round_diff(dsets):
-    dsets[1]['lat'].data = (dsets[1].lat + 0.0000001).data
+    dsets[1]['lat'].data = dsets[1].lat.data + 0.00001
+    with pytest.raises(AssertionError):
+        xr.testing.assert_equal(dsets[0].lat, dsets[1].lat)
     ds = aggregate.concat_ensembles(dsets)
     assert 'member_id' in ds.coords
     assert ds.air.shape == (2, 2920, 25, 53)
     xr.testing.assert_identical(ds.lat, dsets[0].lat)
+    with pytest.raises(AssertionError):
+        xr.testing.assert_equal(ds.lat, dsets[1].lat)
 
 
 def test_drop_additional_coord_dims(dsets):
@@ -66,3 +70,9 @@ def test_drop_additional_coord_dims(dsets):
     assert 'random_var' not in set(ds.variables)
     assert 'random_coord' not in set(ds.variables)
     assert ds.air.shape == (2, 2920, 25, 53)
+
+
+def test_override_coords_mismatch(dsets):
+    dsets[1] = dsets[1].isel(lat=slice(0, 12))
+    with pytest.raises(AssertionError):
+        aggregate.concat_ensembles(dsets)

--- a/tests/test_aggregate.py
+++ b/tests/test_aggregate.py
@@ -53,7 +53,7 @@ def test_concat_ensembles_round_diff(dsets):
     dsets[1]['lat'].data = dsets[1].lat.data + 0.00001
     with pytest.raises(AssertionError):
         xr.testing.assert_equal(dsets[0].lat, dsets[1].lat)
-    ds = aggregate.concat_ensembles(dsets)
+    ds = aggregate.concat_ensembles(dsets, override_coords=True)
     assert 'member_id' in ds.coords
     assert ds.air.shape == (2, 2920, 25, 53)
     xr.testing.assert_identical(ds.lat, dsets[0].lat)
@@ -75,4 +75,4 @@ def test_drop_additional_coord_dims(dsets):
 def test_override_coords_mismatch(dsets):
     dsets[1] = dsets[1].isel(lat=slice(0, 12))
     with pytest.raises(AssertionError):
-        aggregate.concat_ensembles(dsets)
+        aggregate.concat_ensembles(dsets, override_coords=True)

--- a/tests/test_aggregate.py
+++ b/tests/test_aggregate.py
@@ -1,0 +1,58 @@
+import pytest
+import xarray as xr
+
+from intake_esm import aggregate
+
+
+@pytest.fixture
+def dsets():
+    ds = xr.tutorial.open_dataset('rasm', decode_times=False, chunks={'time': 12})
+    ds1 = ds.copy(True)
+    ds2 = ds.copy(True)
+    ds1.attrs['tracking_id'] = 'abc'
+    ds2.attrs['tracking_id'] = 'xyz'
+    return [ds1, ds2]
+
+
+def test_ensure_time_coord_name(dsets):
+    t = aggregate.ensure_time_coord_name(dsets[0], 'time')
+    assert t == 'time'
+
+
+def test_dict_union(dsets):
+    u = aggregate.dict_union(*[ds.attrs for ds in dsets])
+    assert isinstance(u, dict)
+
+    v = aggregate.dict_union(dsets[0].attrs)
+    assert v == dsets[0].attrs
+
+
+def test_merge(dsets):
+    d = dsets[0]
+    d['TS'] = d['Tair']
+    dsets[0] = d
+    ds = aggregate.merge(dsets)
+    assert isinstance(ds, xr.Dataset)
+    assert 'TS' in ds.data_vars
+    xr.testing.assert_identical(ds.time, dsets[0].time)
+
+
+def test_concat_time_levels(dsets):
+    dsets[0]['time'].data = (dsets[0].time + (3 * 365.0)).data
+    ds = aggregate.concat_time_levels(dsets, 'time')
+    assert ds.time.shape == (72,)
+
+
+def test_concat_ensembles(dsets):
+    ds = aggregate.concat_ensembles(dsets)
+    assert 'member_id' in ds.coords
+    assert ds.Tair.shape == (2, 36, 205, 275)
+
+
+def test_concat_ensembles_round_diff(dsets):
+    dsets[1]['xc'].data = (dsets[1].xc + 0.0000001).data
+    ds = aggregate.concat_ensembles(dsets)
+    ds = ds.set_coords(['xc', 'yc'])
+    assert 'member_id' in ds.coords
+    assert ds.Tair.shape == (2, 36, 205, 275)
+    xr.testing.assert_identical(ds.xc, dsets[0].xc)

--- a/tests/test_aggregate.py
+++ b/tests/test_aggregate.py
@@ -6,7 +6,7 @@ from intake_esm import aggregate
 
 @pytest.fixture
 def dsets():
-    ds = xr.tutorial.open_dataset('rasm', decode_times=False, chunks={'time': 12})
+    ds = xr.tutorial.open_dataset('air_temperature', decode_times=False, chunks={'time': 12})
     ds1 = ds.copy(True)
     ds2 = ds.copy(True)
     ds1.attrs['tracking_id'] = 'abc'
@@ -29,30 +29,40 @@ def test_dict_union(dsets):
 
 def test_merge(dsets):
     d = dsets[0]
-    d['TS'] = d['Tair']
+    d['Tair'] = d['air']
     dsets[0] = d
     ds = aggregate.merge(dsets)
     assert isinstance(ds, xr.Dataset)
-    assert 'TS' in ds.data_vars
+    assert 'Tair' in ds.data_vars
     xr.testing.assert_identical(ds.time, dsets[0].time)
 
 
 def test_concat_time_levels(dsets):
-    dsets[0]['time'].data = (dsets[0].time + (3 * 365.0)).data
+    dsets[0]['time'].data = (dsets[0].time + (2 * 2920.0)).data
     ds = aggregate.concat_time_levels(dsets, 'time')
-    assert ds.time.shape == (72,)
+    assert ds.time.shape == (5840,)
 
 
 def test_concat_ensembles(dsets):
     ds = aggregate.concat_ensembles(dsets)
     assert 'member_id' in ds.coords
-    assert ds.Tair.shape == (2, 36, 205, 275)
+    assert ds.air.shape == (2, 2920, 25, 53)
 
 
 def test_concat_ensembles_round_diff(dsets):
-    dsets[1]['xc'].data = (dsets[1].xc + 0.0000001).data
+    dsets[1]['lat'].data = (dsets[1].lat + 0.0000001).data
     ds = aggregate.concat_ensembles(dsets)
-    ds = ds.set_coords(['xc', 'yc'])
     assert 'member_id' in ds.coords
-    assert ds.Tair.shape == (2, 36, 205, 275)
-    xr.testing.assert_identical(ds.xc, dsets[0].xc)
+    assert ds.air.shape == (2, 2920, 25, 53)
+    xr.testing.assert_identical(ds.lat, dsets[0].lat)
+
+
+def test_drop_additional_coord_dims(dsets):
+    dsets[1]['random_coord'] = dsets[1].lat * 0.001
+    dsets[1] = dsets[1].set_coords('random_coord')
+    dsets[1]['random_var'] = xr.DataArray(range(len(dsets[1].lat)), dims=['random_coord'])
+
+    ds = aggregate.concat_ensembles(dsets)
+    assert 'random_var' not in set(ds.variables)
+    assert 'random_coord' not in set(ds.variables)
+    assert ds.air.shape == (2, 2920, 25, 53)

